### ISSUE TITLE
Support patterns in nullable walker

### ISSUE THREAD

**Issue by AlekseyTs** [dotnet/roslyn#29909]
```
        // PROTOTYPE(NullableReferenceTypes): Should not warn on either call to F(string).
        [Fact(Skip = "TODO")]
        public void IsPattern_02()
        {
            var source =
@"class C
{
    static void F(string s) { }
    static void G(string? s)
    {
        if (s is string t)
        {
            F(t);
            F(s);
        }
    }
}";
            var comp = CreateCompilation(new[] { source, NonNullTypesTrue, NonNullTypesAttributesDefinition });
            comp.VerifyDiagnostics();
        }

        // PROTOTYPE(NullableReferenceTypes): Should only warn on F(x) in `case null`.
        [Fact(Skip = "TODO")]
        public void PatternSwitch()
        {
            var source =
@"class C
{
    static void F(object o) { }
    static void G(object? x)
    {
        switch (x)
        {
            case string s:
                F(s);
                F(x); // string s
                break;
            case object y when y is string t:
                F(y);
                F(t);
                F(x); // object y
                break;
            case null:
                F(x); // null
                break;
            default:
                F(x); // default
                break;
        }
    }
}";
            var comp = CreateCompilation(new[] { source, NonNullTypesTrue, NonNullTypesAttributesDefinition });
            comp.VerifyDiagnostics(
                // (18,19): warning CS8604: Possible null reference argument for parameter 'o' in 'void C.F(object o)'.
                //                 F(x); // null
                Diagnostic(ErrorCode.WRN_NullReferenceArgument, "x").WithArguments("o", "void C.F(object o)").WithLocation(18, 19));
        }
```

**Comment by jcouv** [comment]
I expected that `is` patterns would work (https://github.com/dotnet/roslyn/pull/28686)
For nullability in `switch`, I think it's already covered by https://github.com/dotnet/roslyn/issues/23944

**Comment by alrz** [comment]
Another test case:
```cs
if (!(e is string s)) s = null;
var x = s.Length; // warning
```
